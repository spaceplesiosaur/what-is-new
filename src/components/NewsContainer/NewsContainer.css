.newsContainer {
  display: flex;
  flex-flow: wrap;
  width: auto;
  overflow: scroll;
  align-items: center;
  justify-content: center;
  /* width: 79em; */
  width: 100%;
}
