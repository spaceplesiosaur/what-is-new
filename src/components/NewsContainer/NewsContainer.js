import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle.js';

class NewsContainer extends React.Component {
  // constructor(props) {
  //   super(props);
  // }
  renderNewsArticles() {
    return this.props.articleList.map((article, i) => {
      return (
        <NewsArticle
        key={i}
        id={article.name}
        img={article.img}
        headline={article.headline}
        description={article.description}
        url={article.url}
        />
      )
    })
  }
  render() {
    return (
      <section className="newsContainer">
        {this.renderNewsArticles()}
      </section>
    );
  }
}

export default NewsContainer;
