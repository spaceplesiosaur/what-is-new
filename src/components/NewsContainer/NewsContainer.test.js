import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('should match the snapshot with correct data', () => {
    const wrapper = shallow(<NewsContainer
    articleList={['test']}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
