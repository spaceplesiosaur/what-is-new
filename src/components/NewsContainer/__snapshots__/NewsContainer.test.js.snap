// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot with correct data 1`] = `
<section
  className="newsContainer"
>
  <NewsArticle
    key="0"
  />
</section>
`;
